src/utils/UriUtils.ts

```diff
@@ -0,0 +1,42 @@
+export interface Ror2Uri {
+    scheme: "ror2mm";
+    version: string;
+    action: string;
+    parameters: string[];
+}
+
+const ror2UriRe = new RegExp(/ror2mm:\/\/(\w+)\/(\w+)\/?(.*\/?)/);
+
+export const URI_ACTION_INSTALL = "install";
+export const URI_ACTION_SYNC = "sync";
+
+const SUPPORTED: Record<string, string[]> = {
+    "v1": [URI_ACTION_INSTALL, URI_ACTION_SYNC],
+};
+
+
+/**
+ * Helper for parsing and validating custom scheme URIs.
+ *
+ * Deep linking to the mod manager is done via ror2mm:// scheme.
+ */
+export const parseCustomUri = (url: string): Ror2Uri|null => {
+    const matches = url.match(ror2UriRe);
+
+    if (!matches) {
+        return null;
+    }
+
+    const [, version, action, parameters] = matches;
+
+    if (!SUPPORTED[version]?.includes(action)) {
+        return null;
+    }
+
+    return {
+        scheme: "ror2mm",
+        version,
+        action,
+        parameters: parameters.split("/").filter(Boolean)
```

Comment:
the `.filter(Boolean)` looks wrong; wouldn't it make parameter positioning inconsistent? e.g. if you need to populate the 3rd parameter but leave out the 2nd, `foo//bar` would now be parsed as `["foo", "bar"]` instead of the probably intended `["foo", "", "bar"]`?

Reply:
The idea of the filter was to remove empty parameters caused by trailing slash or URI without paramaters at all. Not having to check for empty parameters would make it easier to consume the resulting object,

If we want to support empty parameters, the same effect can be achieved in more verbose ways. But do we want to? In what scenarios would empty parameters (especially in the middle) make sense?

Reply:
I feel like this easily enters the territory of bikeshedding and has no real impact on anything atm, so no need to change anything.

My comment was from the PoV of a future-developer who wants to add/extend the protocol functionality and would have (a very fair assumption) that the positional arguments retain their position. I'm not sure how realistic it is that there would be empty arguments left in, and it's probably not worth analyzing further.

That all said, if the intention is to just get rid of trailing slashes, perhaps there's a less false-positive prone way to do that?

Reply:
@anttimaki marking this as resolved due to the above, up to you if you want to make changes